pkg/gatewayconfigurationserver/managed/grpc_disabled.go

```diff
@@ -0,0 +1,158 @@
+// Copyright © 2024 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package managed
+
+import (
+	"context"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+var errManagedGatewayConfigurationServerDisabled = errors.DefineNotFound(
+	"managed_gateway_configuration_server_disabled", "managed gateway configuration server is disabled",
+)
+
+type disabledManagedGCSServer struct {
+	Component
+	ttnpb.UnsafeManagedGatewayConfigurationServiceServer
```

Comment:
I'm not sure why we use the `UnsafeXX` stubs instead of the `UnimplementedXXX` stubs. This is from the original code @johanstokking.

Reply:
This is fine. Unsafe is causing the code compilation to break and Unimplemented silently returns the Unimplemented error code instead. In this case you want compilation to fail to return the proper error code.